src/accountsdb/accounts_file.zig

```diff
@@ -292,23 +292,29 @@ pub const AccountFile = struct {
 
     pub fn validate(
         self: *const Self,
-        metadata_allocator: std.mem.Allocator,
         buffer_pool: *BufferPool,
     ) !usize {
         var offset: usize = 0;
         var number_of_accounts: usize = 0;
         var account_bytes: usize = 0;
 
+        var buffer_pool_frame_buf: [BufferPool.MAX_READ_BYTES_ALLOCATED]u8 = undefined;
+        var fba = std.heap.FixedBufferAllocator.init(&buffer_pool_frame_buf);
+        const allocator = fba.allocator();
+
         while (true) {
             const account = self.readAccount(
-                metadata_allocator,
+                allocator,
```

Comment:
If this size invariant actually holds, this is a very nice saving, both here and the other couple places you do this.

Reply:
Tested with mainnet, it holds. You just need to read a valid account (10MiB max)